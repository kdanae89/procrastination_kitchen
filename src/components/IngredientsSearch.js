import React from 'react';

class IngredientsSearch extends React.Component {
  state = { term: '' };

  onFormSubmit = (event) => {
    event.preventDefault();
    this.props.onSubmit(this.state.term)
  }

  render () {
    return (
      <div>
        <form
          onSubmit={this.onFormSubmit}
          style={{ marginTop: "1em" }}
        >
          <div
            className="field"
            style={{
              display: "flex",
              justifyContent: "center"
          }}>
            <label>What's in your pantry?</label>
            <input
              type="text"
              style={{ width: '50%'}}
              onChange={(e) => this.setState({ term: e.target.value})}
              value={this.state.term}
            />
          </div>
        </form>
      </div>
    );
  }
}

export default IngredientsSearch;
