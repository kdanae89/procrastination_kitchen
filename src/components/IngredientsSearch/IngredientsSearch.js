import React from 'react';

class IngredientsSearch extends React.Component {
  state = { ingredients: '' };

  onFormSubmit = (event) => {
    event.preventDefault();
    this.props.onSubmit(this.state.ingredients)
  }

  render () {
    return (
      <div>
        <form
          onSubmit={this.onFormSubmit}
          style={{ marginTop: "1em" }}
        >
          <div
            className="field"
            style={{
              display: "flex",
              justifyContent: "center"
          }}>
            <label style={{
              marginRight: "1em"
            }}>
              What's in your pantry?
            </label>
            <input
              data-testid="search"
              type="text"
              style={{ width: '50%'}}
              onChange={(e) => this.setState({ ingredients: e.target.value})}
              value={this.state.ingredients}
            />
          </div>
        </form>
      </div>
    );
  }
}

export default IngredientsSearch;
