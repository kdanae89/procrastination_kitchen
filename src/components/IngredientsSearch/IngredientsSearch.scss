.search-input {
  margin-top: 1em;

  .field {
    display: flex;
    justify-content: center;

    label {
      margin-right: 1em;
    }

    input {
      width: 50%;
    }
  }
}
