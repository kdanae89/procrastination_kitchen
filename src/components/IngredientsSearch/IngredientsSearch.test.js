import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import IngredientsSearch from './IngredientsSearch';

//unit

describe('IngredientSearch', () => {
  it('should not have an initial value', () => {

  })
});
