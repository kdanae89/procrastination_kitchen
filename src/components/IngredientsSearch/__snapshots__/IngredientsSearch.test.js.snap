// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientSearch it renders correctly 1`] = `
<div
  data-testid="search"
>
  <form
    className="search-input"
    data-testid="search-input"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <label>
        What's in your pantry?
      </label>
      <input
        data-testid="input"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </form>
</div>
`;
